compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -1109,6 +1117,63 @@ public BLangNode transform(AnnotationNode annotation) {
         return bLAnnotationAttachment;
     }
 
+    @Override
+    public BLangNode transform(AnnotationDeclarationNode annotationDeclarationNode) {
+        BLangAnnotation annotationDecl = (BLangAnnotation) TreeBuilder.createAnnotationNode();
+        DiagnosticPos pos = getPosition(annotationDeclarationNode);
+        annotationDecl.pos = pos;
+        annotationDecl.name = createIdentifier(pos, annotationDeclarationNode.annotationTag().text());
```

Comment:
I have updated this method to handle missing nodes (Identifiers). PR is yet to be merged (#23669), but better to sync with it, if it got merged before this.

Reply:
Noted

Reply:
Should use `createIdentifier` which accept a `STNode` instead of `String` as the second param.